os/arch/arm/src/amebasmart/amebasmart_i2s.c

```diff
@@ -559,7 +559,16 @@ static void i2s_tx_schedule(struct amebasmart_i2s_s *priv, int result)
 		/* Start next transfer */
 		amebasmart_i2s_tx(priv, bfcontainer);
 	} else if ((priv->apb_tx->nbytes - priv->apb_tx->curbyte) <= 0) {
-		ameba_i2s_pause(priv->i2s_object);
+		if (i2s_dma_tx_done(I2S_DMA_PAGE_NUM) == OK ){
+			/*if all dma tx done, we should pause i2s dma to mute it*/
+			ameba_i2s_pause(priv->i2s_object);
+		} else if (priv->apb_tx->nbytes < I2S_DMA_PAGE_SIZE * I2S_DMA_PAGE_NUM) {
```

Comment:
I think if remained data is greater than 1 slot of dma buffer, it should send to tx immediately.
Otherwise this will occur underrun if next i2s_send delayed. It should consume remained data if there's remained data.

Reply:
If there is pending container to send,  !sq_empty(&priv->tx.pend) will send to tx immediately.
When we reached here, there should not be any remain data in the current container because in i2s_transfer_tx_handleirq is checks for remaining data in container, and only callback when all data are moved to dma.

Reply:
ah, ok I agree with you. Let us test something more, and I will merge if everything work fine. thanks.